src/Components/Artwork/useArtworkLists.ts

```diff
@@ -0,0 +1,82 @@
+import { useToasts } from "@artsy/palette"
+import {
+  SaveArtworkToListsOptions,
+  ResultAction,
+  useSaveArtworkToLists,
+} from "Components/Artwork/SaveButton/useSaveArtworkToLists"
+import { useEffect } from "react"
+import { useTranslation } from "react-i18next"
+import { useAuthIntent } from "Utils/Hooks/useAuthIntent"
+
+export const useArtworkLists = (options: SaveArtworkToListsOptions) => {
+  const { artwork } = options
+  const { sendToast } = useToasts()
+  const { t } = useTranslation()
+  const { value, clearValue } = useAuthIntent()
+  const {
+    isSaved,
+    saveArtworkToLists: saveToLists,
+    openSelectListsForArtworkModal,
+  } = useSaveArtworkToLists(options)
+
+  useEffect(() => {
+    if (value?.objectId !== artwork.internalID) {
+      return
+    }
+
+    if (artwork.isSavedToCustomLists) {
+      // Display select lists for artwork modal if arwork is saved to the custom lists
+      openSelectListsForArtworkModal()
+    } else if (artwork.isSavedToDefaultList) {
+      // Display toast if artwork is already saved to the default list
+      showToastByAction(ResultAction.SavedToDefaultList)
+    } else {
+      // Save artwork to the default list
+      saveArtworkToLists()
+    }
+
+    clearValue()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value, clearValue])
+
+  const showToastByAction = (action: ResultAction) => {
+    if (action === ResultAction.SavedToDefaultList) {
+      sendToast({
+        variant: "success",
+        message: t(
+          "collectorSaves.saveArtworkToLists.toast.artworkSaved.message"
+        ),
+        action: {
+          label: t(
+            "collectorSaves.saveArtworkToLists.toast.artworkSaved.button"
+          ),
+          onClick: openSelectListsForArtworkModal,
+        },
+      })
+
+      return
+    }
```

Comment:
**question**: do you think it would be a good idea to move all this logic to [useSaveArtworkToLists](https://github.com/artsy/force/blob/3e1fc0c1a9d35d33e5b35b3e661cfa8e1b97cee7/src/Components/Artwork/SaveButton/useSaveArtworkToLists.ts#L30)?

Reply:
> we could distinguish the purpose of these hooks better through their naming

yes, totally agree with you. But to tell the truth, I can't think of anything better than `useSaveArtworkToLists` and `useArtworkLists` and therefore I am open to any suggestions 😅